include/soci/row.h

```diff
@@ -43,6 +44,11 @@ class SOCI_DECL row
     row();
     ~row();
 
+#ifdef SOCI_HAVE_CXX11
+    row(row &&other);
+    row &operator=(row &&other);
```

Comment:
Wouldn't just the following work?

```suggestion
    row(row &&other) = default;
    row &operator=(row &&other) = default;
```

Reply:
No, because it would just perform a member-wise move. However, the moved-from object remains in an indeterminate state (not guaranteed to be empty in case of vectors) and since the destructor accesses these objects again, we would run into undefined behavior. Thus, we have to explicitly clear the vectors to being them into a defined state again.

Reply:
Oh, you're right in principle, but this is so, so annoying: we now need to write much more code just to do the same thing that the compiler is almost certainly doing by default anyhow. I also find [Howard's analysis here](https://stackoverflow.com/a/17735913/15275) convincing and it seems to me that as long as we're not using different allocators (and we don't and never will) this can't actually happen in practice.

Am I missing anything here?

Reply:
I wouldn't want to rely on interpreting something like this and rely on it. Especially not as the workaround to simply conform to the standard is so trivial.

Reply:
I disagree. The workaround is verbose and needs maintenance (e.g. will need to be updated if another member is added to the class). Using default implementations is simple and I'm willing to eat my (non existent, but that's a detail) hat if a vector remains non-empty in any existing or future implementation.

Reply:
Alright - your call (and responsibility :P) :shrug: